python/ray/ml/config.py

```diff
@@ -0,0 +1,100 @@
+from dataclasses import dataclass
+from typing import Any, Dict, Optional
+
+from ray.tune import PlacementGroupFactory
+
+
+# Right now, RunConfig is just an arbitrary dict that specifies tune.run
+# kwargs.
+# TODO: After Tuner is implemented, make this into an actual dataclass
+RunConfig = Dict[str, Any]
+
+ScalingConfig = Dict[str, Any]
+
+
+@dataclass
+class _DataParallelScalingConfig:
```

Comment:
Move this to the data parallel class definition?

Reply:
Agreed, that makes this cleaner (see also comments on PR 1/4). My proposal here would be something like:

```

class Trainer(abc.ABC):

    @dataclass
    class ScalingConfig:
        """Documentation for trainer-specific scaling config"""
        pass

    def __init__(self, scaling_config: Union[ScalingConfig, Dict[str, Any]]):
        self._scaling_config = scaling_config

    def _get_scaling_config_class(self, config: Dict) -> ScalingConfig:
        scaling_config = self.ScalingConfig(**self._scaling_config)
        deep_update(scaling_config.__dict__, config.get("scaling_config", {}))
        return scaling_config



class DataParallelTrainer(Trainer):

    @dataclass
    class ScalingConfig:
        num_workers: int = 1
        num_cpus_per_worker: int = 1
        num_gpus_per_worker: int = 0
```